docs/content/sips/006-spin-plugins.md

```diff
@@ -0,0 +1,233 @@
+
+```bash
+Installing plugin deploy with license Apache 2.0 from https://github.com/fermyon/spin-plugin-deploy/releases/download/v0.1.0/spin-plugin-deploy-v0.1.0-macos-aarch64.tar.gz
+For more information, reference the plugin metadata at `https://github.com/fermyon/spin-plugins/plugin-manifests/deploy.json`.
+Are you sure you want to proceed? ('yes'/'no') (default: no)
+```
+
+The plugin will only be installed if a user enters `yes`. Otherwise, the command exits. 
+
+Spin will reference the plugin manifest in order to fetch the plugin binary and install it into the user’s local data directory under a Spin-managed `plugins` subdirectory. The plugin manifest will be stored within a `manifests` subdirectory. 
+
+After installing a plugin, it can be executed directly from the Spin CLI. For example, a plugin named `$name` would be executed by running `spin $name <args>`. Any additional arguments supplied will be passed when executing the associated binary.
+
+ 
+
+**`spin plugin uninstall`**
+
+The `spin plugin uninstall` command uninstalls a plugin named `$name`. 
+
+```bash
+Uninstall a Spin plugin.
+
+USAGE:
+    spin plugin uninstall <name>
+```
+
+**`spin plugin update`**
+
+The `spin plugin update` command updates one or all plugins. If updating a single plugin, the desired version can be specified. By default, plugins are updated to the latest version in the plugins repository. As with `spin plugin install`, the local path or remote addresses to a plugin manifest can be specified.
+
+```bash
+Update one or all installed Spin plugins.
+
+USAGE:
+    spin plugin update [OPTIONS]
+
+OPTIONS:
+	-a, --all        Update all installed plugins (cannot be used with any other option)
+    -p, --plugin     Name of plugin
+    -v, --version    Desired version to update the plugin to. Defaults to latest. 
+    -f, --file       Path to local manifest (mutex with `-u`)
+    -u, --url        Address of remote manifest (mutex with `-f`)
+    -d, --downgrade  Enables downgrading a plugin to an older specified version.
+    -s, --skip-version-compat-check  Instructs Spin to install the plugin even 
+                                     if its manifest specifies that it is 
+                                     incompatible with the current version of Spin
+```
+
+The update will fail if the latest or user-specified version of the plugin is not [compatible with the current version of Spin](#plugin-compatibility). This failure can be overridden with `--skip-version-compat-check`.
```

Comment:
When would skipping the version compatibility check be useful? In theory, if a plugin is incompatible with a particular version of spin, and you install it with `--skip-version-compat-check`, the plugin would be completely useless because it should not be able to work against that version of spin.

If the plugin is not compatible with that version of spin, what scenarios would require a user to skip the compatibility check?

Reply:
In a scenario where Spin has been updated where the version of Spin does not satisfy the compatibility specified on the plugin manifest but does not break anything for the plugin. 

An example would be of a plugin whose manifest states compatibility as `=0.4` and a new spin version `0.5.0` is released. Now even if the spin version change does not break the plugin, the version compatibility check would render it broken. In such cases, if the plugin is provided by the community, the author must be very involved in updating the manifests. Having this as a flag would allow users to override the check and still install it.

Reply:
Wouldn't this be a non-issue if users made plugins use the syntax `>=1.0` instead of `=1.0`? IMO this seems more like "protecting" users from plugin developers not following the semantic versioning convention, which seems like a misstep.

IMO, as a plugin developer, If I explicitly state that my plugin is compatible with `=0.4`, then I expect it to NOT be usable against any version of spin other than spin `>=0.4<0.5`. Full stop.

Reply:
These are good points. I am fine removing the flag, having it declares something as "okay" behavior when it isnt a best practice. The workaround for users would be:
```
wget github.com/fermyon/spin-plugins/plugin_name.json
# locally modify spinCompatibility
spin plugin install --file plugin_name.json
```

Reply:
removed in c775ecdb5617ac8b3e1